### ISSUE TITLE
trial does not wait for the reactor to stop completely between tests/test methods/test suites/test runs/I don't know what

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11099]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1573|
|Type|defect|
|Created|2006-03-14 16:05:39Z|

I added some prints and ran one test with -u  o demonstrate the problem:

```
exarkun@muon:~$ trial -u twisted.web2.dav.test.test_prop.PROP.test_PROPFIND
ADDING A SYSTEM EVENT TRIGGER 'during' 'shutdown' <bound method SelectReactor.crash of <twisted.internet.selectreactor.SelectReactor object at 0xb771a34c>>
ADDING A SYSTEM EVENT TRIGGER 'during' 'shutdown' <bound method SelectReactor.disconnectAll of <twisted.internet.selectreactor.SelectReactor object at 0xb771a34c>>
Test Pass 1
Running 1 tests.
twisted.web2.dav.test.test_prop
  PROP
    test_PROPFIND ... FIRING 'startup'
                                                     [OK]
ADDING A SYSTEM EVENT TRIGGER 'after' 'shutdown' <function <lambda> at 0xb7417fb4>
FIRING 'shutdown'

-------------------------------------------------------------------------------
Ran 1 tests in 1.070s

PASSED (successes=1)
Test Pass 2
Running 1 tests.
twisted.web2.dav.test.test_prop
  PROP
    test_PROPFIND ... FIRING 'startup'
CONTINUING 'shutdown'
RUNNING <bound method SelectReactor.crash of <twisted.internet.selectreactor.SelectReactor object at 0xb771a34c>>
RUNNING <bound method SelectReactor.disconnectAll of <twisted.internet.selectreactor.SelectReactor object at 0xb771a34c>>
RUNNING <function <lambda> at 0xb7417fb4>
                                                     [OK]
ADDING A SYSTEM EVENT TRIGGER 'after' 'shutdown' <function <lambda> at 0xb7417fb4>
FIRING 'shutdown'
Traceback (most recent call last):
  File "/home/exarkun/Projects/Twisted/trunk/bin/trial", line 24, in ?
    run()
  File "/home/exarkun/Projects/Twisted/trunk/twisted/scripts/trial.py", line 371, in run
    test_result = trialRunner.runUntilFailure(suite)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/runner.py", line 503, in runUntilFailure
    result = self.run(test)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/runner.py", line 487, in run
    suite.run(result)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/runner.py", line 177, in run
    TestSuite.run(self, result)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/runner.py", line 117, in run
    test(result)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/runner.py", line 109, in __call__
    return self.run(result)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/runner.py", line 117, in run
    test(result)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/unittest.py", line 342, in __call__
    return self.run(*args, **kwargs)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/unittest.py", line 476, in run
    self._wait(d)
  File "/home/exarkun/Projects/Twisted/trunk/twisted/trial/unittest.py", line 592, in _wait
    raise KeyboardInterrupt()
KeyboardInterrupt
exarkun@muon:~$
```

<details><summary>Searchable metadata</summary>

```
trac-id__1573 1573
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1142352339000000 1142352339000000
changetime__1143513381000000 1143513381000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

Took me a while to figure out why this branch fixes anything.

```suppressWarnings``` doesn't call the function you give it.  Changing the code to use ```runWithWarningsSuppressed``` is probably the right thing to do.  Or changing it to use ```suppressWarnings``` in the intended fashion, at least.  The good news is that it shouldn't matter whether this is done inside ```_bail``` or outside :)

Aside from this, the original issue seems to be resolved, but test_runUntilFailure hangs indefinitely if run with 'trial -u'.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun|
|-|-|

Changed to use runWithWarningsSuppressed.

trial -u works.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

Merge away

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `closed`|
|-|-|

(In [16490]) Actually wait for the reactor to shut down + tests for runUntilFailure. 

Author: jml
Reviewer: exarkun
Fixes [#11099](https://github.com/twisted/twisted/issues/11099)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
